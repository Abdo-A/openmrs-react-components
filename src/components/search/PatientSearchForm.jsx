import React from 'react';
import PropTypes from "prop-types";
import { Field, reduxForm } from 'redux-form';
import { Button, Form, FormGroup, Grid, Row, Col, ControlLabel } from 'react-bootstrap';

const leftTextAlign = {
  textAlign: "left"
};
const colHeight = {
  height: '10px'
};
const inputSearchStyle = {
  width: "-webkit-fill-available",
  height: "30px"
};

const PatientSearchForm = props => {
  const { handleSubmit, pristine, reset, submitting } = props;

  return (
    <Form onSubmit={handleSubmit}>
      <Grid>
        <Row>
          <FormGroup controlId="searchLabel">
            <Col sm={ 3 } componentClass={ControlLabel}>
              Search Patients
            </Col>
          </FormGroup>
        </Row>
        <Row>
          <FormGroup controlId="searchLabel">
            <Col sm={ 3 }>
              <Field
                component="input"
                name="query"
                type="text"
                style={ inputSearchStyle }
                placeholder="enter name or id"
              />
            </Col>
            <Col sm={2} style={ leftTextAlign }>
              <Button
                bsStyle="primary"
                bsSize="small"
                disabled={pristine || submitting}
                type="submit">
                Search
              </Button>
            </Col>
          </FormGroup>
        </Row>
        <Row>
          <Col sm={20} md={20} style={ colHeight }>
            <span><h1>{ '' }</h1></span>
          </Col>
        </Row>

      </Grid>
    </Form>
  );
};

PatientSearchForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  pristine: PropTypes.bool.isRequired,
  reset: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired,
};

export default reduxForm({
  form: 'search-form' // a unique identifier for this form
})(PatientSearchForm);
