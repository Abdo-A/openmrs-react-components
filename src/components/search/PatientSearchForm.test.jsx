import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import PatientSearchForm from './PatientSearchForm';


describe("patientSearchForm", () => {

  const mockStore = configureMockStore();

  it("should render correctly", () => {

    const store = mockStore({});

    const rendered = renderer.create(
      <Provider store={store}>
        <PatientSearchForm/>
      </Provider>
    );

    expect(rendered.toJSON()).toMatchSnapshot();
  });

});
