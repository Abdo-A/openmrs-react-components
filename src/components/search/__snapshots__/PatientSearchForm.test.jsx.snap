// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`patientSearchForm should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className=""
      >
        <label>
          Search: 
        </label>
        <input
          name="query"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className=""
      >
        <div
          className="btn-toolbar"
          role="toolbar"
        >
          <button
            className="btn btn-success"
            disabled={true}
            type="submit"
          >
            Search
          </button>
          <button
            className="btn btn-danger"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Clear Values
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;
